lib/provider/home_screen_provider.dart

```diff
@@ -62,20 +62,18 @@ class HomeScreenProvider extends ChangeNotifier {
   }
 
   void sortPriceDescending() {
-    for (int j = 1; j < productList.length; j++) {
-      double key = double.parse(productList[j].price.amount);
-      Product temp = productList[j];
-      int i = j - 1;
-      productList.asMap().forEach((index, product) {
-        if (index >= 0 && double.parse(product.price.amount) < key) {
-          productList[index + 1] = productList[index];
-          i = index;
-        }
-      });
-      productList[i + 1] = temp;
-    }
+    productList.forEach((product) {
+      double key = double.parse(product.price.amount);
+      int i = productList.indexOf(product) - 1;
 
-    notifyListeners();
+      while (i >= 0 && double.parse(productList[i].price.amount) < key) {
```

Comment:
We should not be having nested loops. It is not efficient. Use the sort function to do this and also for all other sort functions in this provider. 

```import 'package:flutter/foundation.dart';

class User {
  String name;
  int age;

  User(this.name, this.age);

  @override
  toString() {
    return ('$name - $age years old \n');
  }
}

void main() {
  List<User> users = [
    User('Plumber', 10),
    User('John Doe', 50),
    User('Pipi', 4),
    User('Invisible Man', 31),
    User('Electric Man', 34)
  ];

  // Age from small to large
  users.sort((a, b) => a.age.compareTo(b.age));
  debugPrint('Age ascending order: ${users.toString()}');

  // Age from large to small
  users.sort((a, b) => b.age.compareTo(a.age));
  debugPrint('Age descending order: ${users.toString()}');
}